lib/web_ui/lib/src/engine/canvaskit/interval_tree.dart

```diff
@@ -0,0 +1,111 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.12
+part of engine;
+
+// TODO(hterkelsen): Tests.
+/// Associates a [T] with one or more [_UnicodeRange]s.
+class _IntervalTree<T> {
+  final _IntervalTreeNode<T> root;
+
+  _IntervalTree._(this.root);
+
+  factory _IntervalTree.createFromRanges(
+      Map<T, List<_UnicodeRange>> rangesMap) {
+    // Get a list of all the ranges ordered by start index.
+    List<_Interval<T>> intervals = <_Interval<T>>[];
+    for (T key in rangesMap.keys) {
+      for (_UnicodeRange range in rangesMap[key]!) {
+        intervals.add(_Interval<T>(key, range.start, range.end));
+      }
+    }
```

Comment:
nit: `rangesMap.forEach` might produce smaller code (I'm assuming this part is not performance critical as it's executed once during start-up).

Reply:
@yjbanov : have you tried? besides performance, is the code size ok now for forEach?

Reply:
In a local test I saved a whopping 12 bytes by switching to `forEach` 🤣

Reply:
Done.